src/late_multicellular_stage/Scalis.cs

```diff
@@ -36,6 +46,26 @@ public Scalis(IReadOnlyCollection<MulticellularMetaball> metaballs)
             points[i] = point;
             ++i;
         }
+
+        boneMaxSquareDistanceCache = new float[points.Length];
+
+        for (int j = 0; j < points.Length; j++)
+        {
+            if (points[j].Parent == null)
+                continue;
+
+            float maxDistanceFromCenter = (points[j].Position - points[j].Parent!.Position).Length() * 0.5f
```

Comment:
It might be slightly faster to read the points[j] into a local variable once and then reusing it here as this potentially looks up the value 4 times from the list.